pkg/readiness/ready_tracker.go

```diff
@@ -312,79 +325,111 @@ func (t *Tracker) Run(ctx context.Context) error {
 	// Any failure in the errgroup will cancel goroutines in the group using gctx.
 	// The odd one out is the statsPrinter which is meant to outlive the tracking
 	// routines.
-	grp, gctx := errgroup.WithContext(ctx)
-	t.constraintTrackers = syncutil.RunnerWithContext(gctx)
-	close(t.ready) // The constraintTrackers SingleRunner is ready.
+	errChan := make(chan error)
+	wg := &sync.WaitGroup{}
+	t.constraintTrackers = syncutil.NewSingleRunner(errChan)
+	t.dataTrackers = syncutil.NewSingleRunner(errChan)
+	close(t.ready) // The constraintTrackers and dataTrackers SingleRunners are ready.
 
 	if t.mutationEnabled {
-		grp.Go(func() error {
-			return t.trackAssignMetadata(gctx)
-		})
-		grp.Go(func() error {
-			return t.trackAssign(gctx)
-		})
-		grp.Go(func() error {
-			return t.trackModifySet(gctx)
-		})
-		grp.Go(func() error {
-			return t.trackAssignImage(gctx)
-		})
+		wg.Add(4)
```

Comment:
nit: let's just repeat wg.Add(1) -- this will reduce the chance that someone doing copy/paste development will forget to increment the waitgroup.